### PR TITLE
[t2735] prevent dragging problems

### PR DESCRIPTION
**Pull request by secretrobotron**
1. When our dragging starts to occur, block the browser from starting its own by using `e.preventDefault`.
2. A bit of flag cleanup and reorganization of listeners. `draggedOnce` is unnecessary.

### CODE REVIEW

File: src/util/dragndrop.js [hunk 1]
```diff
@@ -139,47 +148,42 @@ define( [ "core/eventmanager", "util/lang", "util/scroll-group" ],
...
-
-      // Let droppable know that it's about to receive one or more items
-      for ( i = droppables.length - 1; i >= 0; --i ) {
-        droppables[ i ].startDrop();
-      }
-
-      for ( i = selectedDraggables.length - 1; i >= 0; --i ) {
-        selectedDraggable = selectedDraggables[ i ];
-        selectedDraggable.stop();
+    // Collect all the droppables
```
Comment (secretrobotron) [inline src/util/dragndrop.js:155]: All of this is really just an out-dent.

File: src/util/dragndrop.js [hunk 2]
```diff
@@ -139,47 +148,42 @@ define( [ "core/eventmanager", "util/lang", "util/scroll-group" ],
...
+    }
 
-      // Let droppable know that we're done dropping
-      for ( i = droppables.length - 1; i >= 0; --i ) {
-        droppables[ i ].stopDrop();
-      }
+    for ( i = selectedDraggables.length - 1; i >= 0; --i ) {
+      selectedDraggable = selectedDraggables[ i ];
+      selectedDraggable.drop();
+    }
```
Comment (dseif) [inline src/util/dragndrop.js:177]: this for loop and the one above look to be the same. Accidental duplicate?

File: src/util/dragndrop.js [hunk 3]
```diff
@@ -139,47 +148,42 @@ define( [ "core/eventmanager", "util/lang", "util/scroll-group" ],
...
 
-      DragNDrop.dispatch( "dropfinished" );
+    for ( i = selectedDraggables.length - 1; i >= 0; --i ) {
+      selectedDraggable = selectedDraggables[ i ];
+      selectedDraggable.reset();
+    }
```
Comment (dseif) [inline src/util/dragndrop.js:182]: also can these 3 ( or 2 ) for loops not be joined into 1 single one? Seems like they are looping over the same data each time.

File: src/util/dragndrop.js (hunk 2 above)
Comment (secretrobotron) [inline src/util/dragndrop.js:177]: Nope. One does `drop`, one does `stop`. They happen like this to make sure everything is in a sane state before moving to the next step.

File: src/util/dragndrop.js (hunk 3 above)
Comment (secretrobotron) [inline src/util/dragndrop.js:182]: Same as above. Everything works like a clock here. Could be optimized later, perhaps, but I'd rather not fiddle with it now. It works as intended, and moving stuff around now might be an unnecessary risk.

I really do appreciate the thorough review, but heed the comment here https://github.com/mozilla/butter/pull/1313/files#L0R151.

**Pull request merged**